reconcile/utils/vault.py

```diff
@@ -160,10 +164,19 @@ def read(self, secret):
         secret_field = secret['field']
         secret_format = secret.get('format', 'plain')
         secret_version = secret.get('version')
-        try:
+
+        kv_version = self._get_mount_version_by_secret_path(secret_path)
+
+        data = None
+        if kv_version == 2:
```

Comment:
Are you sure that read_v2 won't raise an exception when trying to read from a v1 engine? I believe you verified that... just checking anyway :)

Reply:
with this change we are no longer trying to read v1 from v2 or vice versa

Reply:
same as here: https://github.com/app-sre/qontract-reconcile/pull/1414/files#diff-421d17fb36e3edd0412b07a396bcdb5a518bd98d1abfc44d5df8f6072d589540R86-R92